From c35bc7e94f3c2253058c5b3a1d801ca8ee9f6d93 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:56 -0400
Subject: [fs] nfsd: Using free_conn free connection

Message-id: <1403560314-32525-55-git-send-email-bfields@redhat.com>
Patchwork-id: 84477
O-Subject: [Patch RHEL7 054/172] NFSD: Using free_conn free connection
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

Connection from alloc_conn must be freed through free_conn,
otherwise, the reference of svc_xprt will never be put.

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 3f42d2c428c724212c5f4249daea97e254eb0546
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 7415eac..5a9588e 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -2287,7 +2287,8 @@ out:
  if (!list_empty(&clp->cl_revoked))
   seq->status_flags |= SEQ4_STATUS_RECALLABLE_STATE_REVOKED;
 out_no_session:
- kfree(conn);
+ if (conn)
+  free_conn(conn);
  spin_unlock(&nn->client_lock);
  return status;
 out_put_session:
-- 
1.7.1